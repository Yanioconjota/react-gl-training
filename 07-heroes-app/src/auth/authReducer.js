import { types } from "../types/types";

//ejemplo de state
/* const state = {
  name: 'Homero',
  logged: true
}; */

//ejemplo de action
/* const loginAction = {
  type: types.login,
  payload: {
    name: 'Homero',
    email: 'homero@email.com'
  }
}; */

export const authReducer = (state = {}, action) => {
  switch (action.type) {
    case types.login:
      return {
        ...action.payload,
        logged: true
      }
    
    case types.logout:
      return {
        logged: false
      }
  
    default:
      return state;
  }
}