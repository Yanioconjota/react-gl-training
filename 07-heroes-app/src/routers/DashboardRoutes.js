import { Routes, Route } from "react-router-dom";
import { Navbar } from '../components/ui/NavBar';
import { DcScreen } from '../components/dc/DcScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { HeroScreen } from "../components/heroes/HeroScreen";
import { AllHeroesScreen } from "../components/heroes/AllHeroesScreen";

const DashboardRoutes = () => {
  return (
    <>
      <Navbar />
      <div className="container-fluid mt-3">
        <div className="row">
          <div className="col-xs-12">
            <Routes>
              <Route path="marvel" element={<MarvelScreen />} />
              <Route path="dc" element={<DcScreen />} />
              <Route path="all" element={<AllHeroesScreen />} />
              <Route path="search" element={<SearchScreen />} />
              <Route path="hero/:heroId" element={<HeroScreen />} />
              <Route path="/" element={<MarvelScreen />} />
            </Routes>
          </div>
        </div>
      </div>
    </>
  );
};

export default DashboardRoutes;
