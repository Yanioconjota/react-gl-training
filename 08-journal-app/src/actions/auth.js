import Swal from "sweetalert2";
import { types } from "../types/types";
import {
  googleAuthProvider,
  getAuth,
  signInWithPopup,
  signInWithEmailAndPassword,
  createUserWithEmailAndPassword,
  signOut,
  updateProfile,
} from "../firebase/firebase-config";
import { finishLoading, startLoading } from "./ui";
import { notesLogout } from "./notes";

//gracias al midleware thunk-redux tenemos acceso al dispatch dentro del return
export const startLoginWithEmailPassword = (email, password) => {
  return (dispatch) => {
    const auth = getAuth();
    dispatch(startLoading());
    signInWithEmailAndPassword(auth, email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch(finishLoading());
      })
      .catch((err) => {
        dispatch(finishLoading());
        //mensaje de error en modal
        Swal.fire('Error', err.message, 'error');
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    const auth = getAuth();
    dispatch(startLoading());
    createUserWithEmailAndPassword(auth, email, password)
      .then(async ({ user }) => {
        //update profile agrega el displayName, ya que este solo se crea cuando nos registramos con google o cualquier red social
        await updateProfile(user, { displayName: name });
        dispatch(login(user.uid, user.displayName));
        dispatch(finishLoading());
      })
      .catch((err) => {
        dispatch(finishLoading());
        //mensaje de error en modal
        Swal.fire("Error", err.message, "error");
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    const auth = getAuth();
    signInWithPopup(auth, googleAuthProvider).then(({ user }) => {
      dispatch(login(user.uid, user.displayName));
    });
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const startLogout = () => {
  return async (dispatch) => {
    const auth = getAuth();
    await signOut(auth);
    dispatch(logout());
    dispatch(notesLogout());
  };
};

export const logout = () => ({
  type: types.logout,
});
