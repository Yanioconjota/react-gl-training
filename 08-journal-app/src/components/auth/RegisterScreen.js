import React from 'react';
import { Link } from "react-router-dom";
import { useForm } from '../../hooks/useForm';

export const RegisterScreen = () => {

  const [formValues, handleInputChange] = useForm({
    name: 'Homero',
    email: "homerothompson@gmail.com",
    password: "123456",
    password2: "123456"
  });

  const { name, email, password, password2 } = formValues;

  const handleRegister = (e) => {
    e.preventDefault();
    console.log(name, email, password, password2);
  }

  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form onSubmit={ handleRegister }>
        <input type="text"
               placeholder="Name"
               name="name"
               value={ name }
               onChange={ handleInputChange }
               className="auth__input"
               autoComplete="off"/>

        <input type="text"
               placeholder="Email"
               name="email"
               value={ email }
               onChange={ handleInputChange }
               className="auth__input"
               autoComplete="off"/>

        <input type="password" 
               placeholder="Password"
               name="password"
               value={ password }
               onChange={ handleInputChange }
               className="auth__input"/>

        <input type="password" 
               placeholder="Confirm password"
               name="password2"
               value={ password2 }
               onChange={ handleInputChange }
               className="auth__input"/>

        <button type="submit"
                className="btn btn-primary btn-block mb-5">Login
        </button>
        <Link className="link" to="/auth/login">
          Already registered?
        </Link>
      </form>
    </>
  )
}
