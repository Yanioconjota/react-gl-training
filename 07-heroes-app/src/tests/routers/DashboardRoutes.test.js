import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import DashboardRoutes from "../../routers/DashboardRoutes";

describe('Pruebas con DashboardRoutes', () => {

  const contextValue = {
    user: {
      logged: true,
      name: 'Homero'
    }
  };
  test('debe mostrarse correctamente la ruta por defecto (Marvel)', () => {
    
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        {/* Permite hacer pruebas en componentes que hacen uso del useNavigate */}
        <MemoryRouter initialEntries={ ['/'] }>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info > span').text().trim()).toBe('Homero');
    expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen');
  });

  test('debe mostrarse correctamente la ruta DC', () => {
    
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        {/* Permite hacer pruebas en componentes que hacen uso del useNavigate */}
        <MemoryRouter initialEntries={ ['/dc'] }>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('DC Screen');
  });
});