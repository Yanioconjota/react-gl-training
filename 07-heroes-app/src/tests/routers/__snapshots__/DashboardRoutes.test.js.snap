// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con DashboardRoutes debe mostrarse correctamente la ruta DC 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <strong>
          HEROAPP
        </strong>
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            activeclassname="active"
            aria-current="page"
            className="nav-item nav-link active"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/all"
            onClick={[Function]}
          >
            Todos
          </a>
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Buscar
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item nav-link text-info"
          >
            <span>
              Homero
            </span>
          </li>
          <li
            className="nav-item nav-link"
            onClick={[Function]}
          >
            Logout
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container-fluid mt-3"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        Array [
          <h1>
            DC Screen
          </h1>,
          <hr />,
          <div
            className="row
 rows-cols-1
 row-cols-md-3 g-3
 animate__animated animate__fadeIn"
          >
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Batman"
                  className="card-img-top"
                  src="./assets/img/dc-batman.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Batman
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-batman"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Black Canary"
                  className="card-img-top"
                  src="./assets/img/dc-black.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Black Canary
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-black"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Blue Beetle"
                  className="card-img-top"
                  src="./assets/img/dc-blue.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Blue Beetle
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-blue"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Flash"
                  className="card-img-top"
                  src="./assets/img/dc-flash.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Flash
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-flash"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Green Arrow"
                  className="card-img-top"
                  src="./assets/img/dc-arrow.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Green Arrow
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-arrow"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Green Lantern"
                  className="card-img-top"
                  src="./assets/img/dc-green.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Green Lantern
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-green"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Martian Manhunter"
                  className="card-img-top"
                  src="./assets/img/dc-martian.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Martian Manhunter
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-martian"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Robin/Nightwing"
                  className="card-img-top"
                  src="./assets/img/dc-robin.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Robin/Nightwing
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-robin"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Superman"
                  className="card-img-top"
                  src="./assets/img/dc-superman.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Superman
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-superman"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Wonder Woman"
                  className="card-img-top"
                  src="./assets/img/dc-wonder.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Wonder Woman
                  </h2>
                  <p>
                    DC Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/dc-wonder"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
          </div>,
        ]
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas con DashboardRoutes debe mostrarse correctamente la ruta por defecto (Marvel) 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <strong>
          HEROAPP
        </strong>
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/all"
            onClick={[Function]}
          >
            Todos
          </a>
          <a
            activeclassname="active"
            className="nav-item nav-link"
            href="/search"
            onClick={[Function]}
          >
            Buscar
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item nav-link text-info"
          >
            <span>
              Homero
            </span>
          </li>
          <li
            className="nav-item nav-link"
            onClick={[Function]}
          >
            Logout
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container-fluid mt-3"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12"
      >
        Array [
          <h1>
            Marvel Screen
          </h1>,
          <hr />,
          <div
            className="row
 rows-cols-1
 row-cols-md-3 g-3
 animate__animated animate__fadeIn"
          >
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Captain America"
                  className="card-img-top"
                  src="./assets/img/marvel-captain.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Captain America
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Cyclops"
                  className="card-img-top"
                  src="./assets/img/marvel-cyclops.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Cyclops
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Daredevil"
                  className="card-img-top"
                  src="./assets/img/marvel-daredevil.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Daredevil
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Hawkeye"
                  className="card-img-top"
                  src="./assets/img/marvel-hawkeye.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Hawkeye
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Hulk"
                  className="card-img-top"
                  src="./assets/img/marvel-hulk.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Hulk
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Iron Man"
                  className="card-img-top"
                  src="./assets/img/marvel-iron.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Iron Man
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Silver Surfer"
                  className="card-img-top"
                  src="./assets/img/marvel-silver.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Silver Surfer
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Spider-Man"
                  className="card-img-top"
                  src="./assets/img/marvel-spider.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Spider-Man
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Thor"
                  className="card-img-top"
                  src="./assets/img/marvel-thor.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Thor
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
            <div
              className="card col-6 col-md-3 col-lg-2"
            >
              <div
                className="row no-gutters"
              >
                <img
                  alt="Wolverine"
                  className="card-img-top"
                  src="./assets/img/marvel-wolverine.jpg"
                />
                <div
                  className="card-body px-4"
                >
                  <h2
                    className="card-title"
                  >
                    Wolverine
                  </h2>
                  <p>
                    Marvel Comics
                  </p>
                  <a
                    className="btn w-100 mt-3 btn-info"
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Mas info...
                  </a>
                </div>
              </div>
            </div>
          </div>,
        ]
      </div>
    </div>
  </div>,
]
`;
